Add OASA Telematics greek public transport sensor component

panosmz opened the pull request:
## Description:

**Pull request in [home-assistant.io](https://github.com/home-assistant/home-assistant.io) with documentation (if applicable):** <https://github.com/home-assistant/home-assistant.io/pull/8836>

## Example entry for `configuration.yaml` (if applicable):
```yaml
sensor:
  - platform: oasa_telematics
    route_id: '090006'
    stop_id: '1965'
```

## Checklist:
  - [X] The code change is tested and works locally.
  - [X] Local tests pass with `tox`. **Your PR cannot be merged unless tests pass**
  - [X] There is no commented out code in this PR.

If user exposed functionality or configuration variables are added/changed:
  - [X] Documentation added/updated in [home-assistant.io](https://github.com/home-assistant/home-assistant.io)

If the code communicates with devices, web services, or third-party tools:
  - [X] New dependencies have been added to the `REQUIREMENTS` variable ([example][ex-requir]).
  - [X] New dependencies are only imported inside functions that use them ([example][ex-import]).
  - [X] New or updated dependencies have been added to `requirements_all.txt` by running `script/gen_requirements_all.py`.
  - [X] New files were added to `.coveragerc`.

fronzbot committed:
Upgrade blinkpy==0.13.1 (Fixes #21559) (#21578)

* Upgrade blinkpy with new api endpoint

* Change wifi units to dBm

panosmz committed:
Add OASA Telematics greek public transport sensor component

panosmz committed:
fixed violations

panosmz committed:
fix lint errors

homeassistant commented:
Hello @panosmz,

When attempting to inspect the commits of your pull request for CLA signature status among all authors we encountered commit(s) which were not linked to a GitHub account, thus not allowing us to determine their status(es).

The commits that are missing a linked GitHub account are the following:

- [`8d28bb94827ca1bf9127cf632b5206f7ac51ce88`](https:&#x2F;&#x2F;github.com&#x2F;home-assistant&#x2F;home-assistant&#x2F;pull&#x2F;21697&#x2F;commits&#x2F;8d28bb94827ca1bf9127cf632b5206f7ac51ce88) - This commit has something that looks like an email address (panosmz@outlook.com). Maybe try linking that to GitHub?.

Unfortunately, **we are unable to accept this pull request until this situation is corrected.**

Here are your options:

1. If you had an email address set for the commit that simply wasn't linked to your GitHub account you can link that email now and it will retroactively apply to your commits. The simplest way to do this is to click the link to one of the above commits and look for a blue question mark in a blue circle in the top left. Hovering over that bubble will show you what email address you used. Clicking on that button will take you to your email address settings on GitHub. Just add the email address on that page and you're all set. GitHub has more information about this option [in their help center](https://help.github.com/articles/setting-your-email-in-git/#commits-on-github-arent-linking-to-my-account).

2. If you didn't use an email address at all, it was an invalid email, or it's one you can't link to your GitHub, you will need to change the authorship information of the commit and your global Git settings so this doesn't happen again going forward. GitHub provides some great instructions on how to change your authorship information [in their help center](https://help.github.com/articles/setting-your-email-in-git/).
   * If you only made a single commit you should be able to run
      ```
      git commit --amend --author="Author Name <email@address.com>"
      ```
      (substituting Author Name and email@address.com for your actual information) to set the authorship information.
   * If you made more than one commit and the commit with the missing authorship information is not the most recent one you have two options:
       1. You can re-create all commits missing authorship information. This is going to be the easiest solution for developers that aren't extremely confident in their Git and command line skills.
       2. You can use [this script](https://help.github.com/articles/changing-author-info/) that GitHub provides to rewrite history. **Please note:** this should be used only if you are very confident in your abilities and understand its impacts.
   * Whichever method you choose, I will come by to re-check the pull request once you push the fixes to this branch.

We apologize for this inconvenience, especially since it usually bites new contributors to Home Assistant. We hope you understand the need for us to protect ourselves and the great community we all have built legally. The best thing to come out of this is that you only need to fix this once and it benefits the entire Home Assistant and GitHub community.

Thanks, I look forward to checking this PR again soon! :heart:

panosmz commented:
I added the missing email address. Cla-error should be fixed.

panosmz committed:
added @staticmethod decorator to empty_name_data

panosmz committed:
fix bad-staticmethod-argument

panosmz commented:
@homeassistant cla-error should be resolved

awarecan committed:
Make pytest in tox quite (#21727)

ljmerza committed:
tplink - catch SmartDeviceException on is_dimmable call (#21726)

* automated commit 06/03/2019 20:49:50

* automated commit 06/03/2019 20:53:13

* automated commit 06/03/2019 20:53:48

Commit:
Fix Name of Homematic IP accesspoint in devices, if name is configured (#21617)

* Fix Name of Accesspoint if name is configured

* fix lint

* Simplyfied naming

* applied suggestion

Co-Authored-By: SukramJ <markus@mm-jankowski.de>

* update comment

Commit:
Fix Name of Homematic IP accesspoint in devices, if name is configured (#21617)

* Fix Name of Accesspoint if name is configured

* fix lint

* Simplyfied naming

* applied suggestion

Co-Authored-By: SukramJ <markus@mm-jankowski.de>

* update comment

Commit:
Fix group-switch availability for Homematic IP (#21640)

* Add available=True to groups

* Added unreach to stateattributes

* Fixed comments

* added missing sabotage check

* added missing lowBat check

* fix typo

* apply suggestion

Co-Authored-By: SukramJ <markus@mm-jankowski.de>

* apply suggestion

Co-Authored-By: SukramJ <markus@mm-jankowski.de>

* applied suggiestions

* readded lost str()

* fix comment

Commit:
Fix group-switch availability for Homematic IP (#21640)

* Add available=True to groups

* Added unreach to stateattributes

* Fixed comments

* added missing sabotage check

* added missing lowBat check

* fix typo

* apply suggestion

Co-Authored-By: SukramJ <markus@mm-jankowski.de>

* apply suggestion

Co-Authored-By: SukramJ <markus@mm-jankowski.de>

* applied suggiestions

* readded lost str()

* fix comment

awarecan committed:
Remove pytest warning message (#21713)

pvizeli committed:
Update .travis.yml (#21736)

* Update .travis.yml

* Update tox.ini

* Update main.workflow

* Update tox.ini

pvizeli committed:
Change github trigger type

pvizeli committed:
Revert Travis until github actions work better for PR (#21746)

pvizeli committed:
Revert pull request to push

zewelor committed:
Allow light toggle service to accept all turn on params (#20912)

panosmz committed:
switch api library to Python: 3.5.3 compatible

panosmz committed:
switch api library to Python: 3.5.3 compatible

balloob committed:
Updated frontend to 20190305.1

dshokouhi committed:
Fix botvac connected maps call as it is not a supported model (#21752)

dshokouhi committed:
Fix botvac connected maps call as it is not a supported model (#21752)

ljmerza committed:
automated commit 07/03/2019 10:47:38 (#21749)

ljmerza committed:
automated commit 07/03/2019 10:47:38 (#21749)

awarecan committed:
Fix colorlog import error (#21754)

* Fix colorlog import error

* Lint

awarecan committed:
Fix colorlog import error (#21754)

* Fix colorlog import error

* Lint

panosmz committed:
updated api library version

panosmz committed:
updated api library version

panosmz committed:
sort time as int instead of string

panosmz committed:
sort time as int instead of string

panosmz committed:
blank line at end of file

panosmz committed:
blank line at end of file

panosmz committed:
blank line at end of file

panosmz committed:
blank line at end of file

panosmz committed:
blank line at end of file

panosmz committed:
blank line at end of file

marvin-w committed:
Introduce target_temperature_state_address for climate device (#21541)

syssi committed:
Bump PyXiaomiGateway version to 0.12.2 (Closes: #21731) (#21764)

syssi committed:
Bump PyXiaomiGateway version to 0.12.2 (Closes: #21731) (#21764)

Commit:
Updated to newest pyeconet (#21772)

Commit:
Updated to newest pyeconet (#21772)

dthulke committed:
adds missing SUPPORT_VOLUME_SET flag to webos media_player (#21766)

dthulke committed:
adds missing SUPPORT_VOLUME_SET flag to webos media_player (#21766)

awarecan committed:
Fix script load module issue (#21763)

* Fix script load depedency

* Revert #21754

awarecan committed:
Fix script load module issue (#21763)

* Fix script load depedency

* Revert #21754

balloob committed:
Bumped version to 0.89.1

balloob committed:
Add workflow for tests

scop committed:
Upgrade pylint to 2.3.1 (#21789)

balloob committed:
Merge pull request #21778 from home-assistant/rc

0.89.1

josemotta committed:
Add Time of Flight Sensor using VL53L1X (#21230)

* Add Time of Flight Sensor using VL53L1X

* Fix issues found by bot

* Fix issues from bot

* Remove extra logs

* Keep removing logs dependencies not used

* Remove log from update

* Add logger info to async_update

* Fix over-indented line

* Fix pylint error

* Remove logger reporting successful operation

* Update requirements

* Update requirements_all.txt

* Update requirements_test_all.txt

* Used isort to keep imports and added STMicroelectronics to docstring

* Replace time.sleep by asyncio.sleep

* Add requirements to COMMENT_REQUIREMENTS and fix typo

* Using async_add_executor_job to schedule the call in the pool

* Fix typo

* Optimize async_update

* Updated requirements files

* Group and schedule calls that should be run sequentially

* Fix lint errors

* Revision showing development history

* Cleaning and typos

* Cleaning and typos

* Fix wrong-import-order

* Fix gen_requirements_all

* Schedule rpi_gpio I/O on the executor thread pool

* Fix partial parameters

* Fix bot error - add blank line

* Fix lint error

* Remove dependencies from requirements

* Review initial commits

* Move all device I/O to async_update

* Update requirements_all.txt

* Revised header with no url to the docs

* Use async_added_to_hass to add and initialize the sensor

* Add docstring to init()

* Move sensor.open() to async_setup_platform

* Remove logging and async

* Fix typo

* Move sensor.open to safe initialization

* Fix typo

* Fix typo

* Add the new tof module to .coveragerc

* Move the sensor platform under a tof package

* Update .coveragerc and requirements_all for tof package

scop committed:
Upgrade huawei-lte-api to 1.1.5 (#21791)

fbradyirl committed:
Adding enigma2 media player (#21271)

* Updated based on review comments

* fix hound

* Update homeassistant/components/media_player/enigma2.py

* Update homeassistant/components/media_player/enigma2.py

* Update homeassistant/components/media_player/enigma2.py

* Update enigma2.py

* Update enigma2.py

* Move file and update docsstring

* Fix path in coverage rc file

* requirements

* Update media_player.py

* Setup discovery for e2

* Handle discovered devices

* Add reqs

* Update for auth for openwebif

* Forget to set DEFAULT_PASSWORD

* Add source selection

* Fix get current source name

* Update pip version

* - adding some extra attributes
- support better recording playback integration

* bump pip version

* Bump pip

* Adding prefer_picon config option

* Updates to move logic into pypi module

* bump pip

* bump pip

* - remove http dependancy.

- rename prefer_picon to use_channel_icon

* Bump pypi to fix toggle bug.

also fix travis

also move setup out of init

pavoni committed:
Bump loopenergy to 0.1.0. Loop updated their socket.io server from 0.9 to 2.0 - which required a library update. (#21809)

nudded committed:
remove occupancy, as it is not available at this level in the iRail api (#21810)

balloob committed:
Onboarding to generate auth code (#21777)

balloob committed:
Remove stub from config component (#21822)

uchagani committed:
change paths to be relative (#21827)

balloob committed:
Change how we import config modules (#21824)

awarecan committed:
Load logger and system_log components as soon as possible (#21799)

awarecan committed:
Resolve auth_store loading race condition (#21794)

* Add lock in auth_store._async_load()

* Python 3.5 does not like assert_called_once()

robbiet480 committed:
Log if aiohttp hits error during IndieAuth (#21780)

* Log if aiohttp hits error during IndieAuth

* Add content of redirect_url into error log

Co-Authored-By: awarecan <awarecan@users.noreply.github.com>

fbradyirl committed:
Add support for Cisco Mobility Express (#21531)

* Move cisco me to new layout

* Add docstring

* Move items out of init method and pass the controller instance to the scanner in get_scanner

* Update homeassistant/components/cisco_mobility_express/device_tracker.py

Co-Authored-By: fbradyirl <fbradyirl@users.noreply.github.com>

* Update homeassistant/components/cisco_mobility_express/device_tracker.py

Co-Authored-By: fbradyirl <fbradyirl@users.noreply.github.com>

* Update homeassistant/components/cisco_mobility_express/device_tracker.py

Co-Authored-By: fbradyirl <fbradyirl@users.noreply.github.com>

* Update homeassistant/components/cisco_mobility_express/device_tracker.py

Co-Authored-By: fbradyirl <fbradyirl@users.noreply.github.com>

* Fix build error

* Cleanup based on comments.

cmsimike committed:
Synology sensor quick return if attr is null (#21709)

* Quick return if attr is null

There are some case where attr is null. Returning null doesn't change anything (in my case this is mapped to a volume that doesn't exist, not sure what others are seeing). 

If you have confirmed you hass instance for C instead of F you do not see this error.

* update == to is

* whitespace

engrbm87 committed:
fix empty TOPIC_BASE issue (#21740)

* fix empty TOPIC_BASE issue

if the value of the TOPIC_BASE is empty then we need to remove "~" from the topic value if it exists. 
by doing `if base:` on line 239 the condition will be false if the value is empty so the '~' will not be stripped from the topic value.
I simply removed the `if base:` line and added `if TOPIC_BASE in payload:`

* Update homeassistant/components/mqtt/discovery.py

Co-Authored-By: engrbm87 <engrbm87@gmail.com>

Commit:
Updated to pyeconet 0.0.10 (#21837)

bachya committed:
Fix config entry exception in Ambient PWS (#21836)

andrewsayre committed:
Update dependencies to receive data on webhook callbacks (#21838)

andrewsayre committed:
Update dependencies to receive data on webhook callbacks (#21838)

bachya committed:
Fixed a misspelling in a docstring (#21846)

robbiet480 committed:
mobile_app improvements (#21607)

* First webhook commands for getting and deleting single registrations

* Keep a list of deleted webhook IDs so we can 410 if the webhook receives traffic in the future

* Return a empty JSON object instead of None

* Split up mobile_app bits into individual files

* Add typing

* Sort keys

* Remove unused async_setup_entry

* New decorator method of registering webhooks

* Add tests for cloud hook forwarding and improve error handling for cloud hooks

* Initial implementation of platform specific logic

* Add get registrations by user ID websocket call, minor style fixes

* Stop using resp dictionary during registration

* Move mobile_app/ios.py to ios/mobile_app.py

* Log any errors encountered during webhook

* Improve update registration call

* Split up mobile_app tests to match split up component

* Fix tests

* Remove integration_map in favor of component name in registration

* Add a few helper functions for custom logic components to use

* Load the app_component platform at device registration or component setup time

* Remove extraneous function

* Use guard function for checking if component is in device

* Inline websocket schemas

* Rename ATTR_s used in storage to DATA_ prefix

* squash flake8 and pylint issues

* Remove ios.mobile_app platform

* Dont mark websocket_api as a dependency

* Return standard empty_okay_response with 400 if no JSON sent

* Ensure deleted webhook IDs are registered at launch

* Remove the creation of cloudhooks during handle_webhook

* Rename device to registration everywhere applicable

* Dont check if cloud is logged in, just check if cloud is in components

* Dont ever use cloudhook_id

* Remove component loading logic for a later PR

* Cast exception to string

* Remove unused functions

awarecan committed:
Override http.trusted_networks by auth_provider.trusted_networks (#21844)

awarecan committed:
Override http.trusted_networks by auth_provider.trusted_networks (#21844)

Hackashaq666 committed:
Update honeywell.py to read current humidity for US Thermostats (#21728)

* Update honeywell.py

Add thermostat humidity reading available in somecomfort for US thermostats.

* Update honeywell.py

* Update honeywell.py

wburgers committed:
Fix TypeError (#21734)

* timediff is of type timedelta. Divide by timedelta does not work.

- convert a timedelta to int
- make sure the test inputs real timestamps

* Convert the total_seconds to decimal and round the result

readings are of type Decimal, so fix test to reflect that

* split line into multiple statements

Line too long

* use total_seconds instead of timediff

* Make both values float instead of Decimal

emontnemery committed:
Introduce Entity.async_write_ha_state() to not miss state transition (#21590)

* Copy state in schedule_update_ha_state

* Lint

* Fix broken test

* Review comment, improve docstring

* Preserve order of state updates

* Rewrite

* Break up async_update_ha_state

* Update binary_sensor.py

* Review comments

* Update docstring

* hass -> ha

* Update entity.py

* Update entity.py

piitaya committed:
Fix authorization header in cors (#21662)

* Fix authorization headers in cors

* Use aiohttp authorization header instead of custom const

awarecan committed:
Only commit if need. (#21848)

dmulcahey committed:
Bump quirks for ZHA and handle resulting battery % change (#21869)

* bump quirks and handle battery change

* move inside guard

* round battery

balloob committed:
Better cloud check (#21875)

zemerick1 committed:
Add ClearPass Policy Manger device tracker (#21673)

* Adding ClearPass Policy Manger device tracker. Amending author

* Cleaned redundant code

* Updated .coveragerc

* Updated requirements_all.txt

* Implemented suggested changes partially.

* Implemented more suggested changes.

* Hound was unhappy

* Implement further suggested changes.

* Make Hound happy.

* Satisfy Travic CI

* Satisfy Travis CI #2

* Hound barking

* pylint else: return

* Implemented suggested changes minus AccessToken

* Removed access token logging

* Removed throttle import

* Removed period from debug string

* Make travis happy :(

* Moved source to new component structure.

* Forgot to rename source.

dshokouhi committed:
Fix botvac when no map exists (#21877)

dshokouhi committed:
Fix botvac when no map exists (#21877)

andrewsayre committed:
Add SmartThings climate support for Air Conditioners (#21840)

* Add support for Samsung Air Conditioners to Climate

* Changes per review feedback.

* Fix py3.5 compat list sorting in test

andrewsayre committed:
Add SmartThings sensor support for Three Axis (#21841)

* Add support for Three Axis to sensor platform

* Changes per review feedback.

* Remove unnecessary KeyError except

* Fix lint issue in line wrapping

balloob committed:
Fix incorrect 2nd param

balloob committed:
Add user group (#21832)

* Add user group

* Rename system group to plural

dmulcahey committed:
Update ZHA state handling (#21866)

* make device available if it was seen within 2 hours

* more state restore

* cleanup init

* clean up storage stuff

* fix tests

* update state handling

balloob committed:
Update translations

balloob committed:
Updated frontend to 20190309.0

ludeeus committed:
Change lib for whois sensor (#21878)

* Change lib for whois sensor

* Change requirements.txt

Julius2342 committed:
Don't wait until final position of Velux cover is reached (#21558)

## Description:

* Bump version to latest version of pyvlx: 0.2.10. Library more failure tolerant, when detecting an unsupported device.
* When calling API (e.g. run scene, change position) don't wait until device has reached target position (This caused  HASS to be flaky while the device was moving)
* Support for vertical and horizontal awnings.

ToRvaLDz committed:
Fix missing code_required check in async_alarm_arm_night (#21858)

* Fix missing code_required check in async_alarm_arm_night

* Remove double code validation
Test added

thibmaek committed:
Fix icon for sensor.discogs_random_record (#21891)

* Fix icon for sensor.discogs_random_record

* Add myself to CODEOWNERS for Discogs code

thibmaek committed:
Add myself to CODEOWNERS for NMBS code (#21892)

* Add myself to CODEOWNERS for NMBS code

* Fix

gertdb committed:
Modbus write_register accept single value and array (#21621)

* Update __init__.py

* Update services.yaml

epleypa committed:
Add custom holidays to workday sensor (#21718)

* Add custom holidays to workday sensor

* correcting copy/paste errors

* resolve block comment and add default

* resolve line too long

* fixed handling of no custom holidays

* hound fixes

* rerun Travis

* @fabaff requested changes

* spaces v tabs

* Fix validation

ndonegan committed:
Return time based attributes as datetime in Unifi module (#21146)

* Return time based data as datetime in Unifi module

* Fix missing space and pylint complaints about import order.

chilicheech committed:
Allow emulated hue to set climate component temperature (#19034)

fbradyirl committed:
bump netdisco to 2.4.0 (#21914)

awarecan committed:
Deprecate http.api_password (#21884)

* Deprecated http.api_password

* Deprecated ApiConfig.api_password

GitHub Drafted PR would trigger CI after changed it to normal PR.
I have to commit a comment change to trigger it

* Trigger CI

* Adjust if- elif chain in auth middleware

emontnemery committed:
Don't hang forever if manually added cast is down (#21565)

* Don't hang forever if manually added cast is down

* Adapt to pychromecast

* Do not set available until connected

* Update __init__.py

* Update requirements

* Lint, tests

* Fix tests

balloob committed:
Add WS subscription command for MQTT (#21696)

* Add WS subscription command for MQTT

* Add test

* Add check for connected

* Rename event_listeners to subscriptions

cisasteelersfan committed:
Add 'ssl' parameter for FiOS Quantum Gateway and upgrade Pypi (#21669)

* bump pypi version and take 'use_https' parameter

* changed to use CONF_SSL

rbdixon committed:
Add as_timestamp() to Jinja filters. (#21910)

Add as_timestamp() to Jinja filters

Jc2k committed:
HomeKit controller config flow fixes (#21898)

* HomeKit controller config flow fixes

* This does work if you have latest vol-serialize

fbradyirl committed:
Update enigma2 based on review comments (#21890)

* Updates based on review comments

* bump netdisco and remove enigma2 init code

* revert netdisco bump

amelchio committed:
Remove confusing warning for TTS without entity_id (#21927)

gertdb committed:
Fixes Modbus service.yaml validity (#21923)

* Update __init__.py

* Update services.yaml

* Update services.yaml

andrewsayre committed:
Add cloudhook support to SmartThings component (#21905)

* Add support for Nabu Casa cloudhooks

* Added tests to cover cloudhook creation and removal

* Remove cloud dependency

robbiet480 committed:
If registration supports encryption then return encrypted payloads (#21853)

michaelarnauts committed:
Allow inverting netdata sensor values (#21711)

* Allow inverting netdata sensor values

* Fix lint issue

* Use parentheses

Oro committed:
Changed from nanoleaf_aurora to nanoleaf (#21913)

Nanoleaf component now supports both nanoleaf lights, Aurora and Canvas
Changed the dependency to pynanoleaf, nanoleaf does not seem to be
maintained anymore

nugget committed:
Minor version bump for anthemav package (#21932)

Additional Python 3.7 fixes in the anthemav package.
See also: https://pypi.org/project/anthemav/

balloob committed:
Add area permission check (#21835)

Jc2k committed:
Add an asyncio Lock around pairing, which cant be used concurrently (#21933)

balloob committed:
Add update user command (#21922)

* Add update user command

* Add is_admin to current user

pvizeli committed:
Offload Cloud component (#21937)

* Offload Cloud component & Remote support

* Make hound happy

* Address comments

fabaff committed:
Upgrade schiene to 0.23 (#21940)

kstaniek committed:
Fixes issues #21821 and #21819 (#21911)

* Fix #21821

* datetime fix

* local time to utc conversion fix

* Test cases update

* date import removed

* Update tod.py

kstaniek committed:
Fixes issues #21821 and #21819 (#21911)

* Fix #21821

* datetime fix

* local time to utc conversion fix

* Test cases update

* date import removed

* Update tod.py

hunterjm committed:
Live Streams Component (#21473)

* initial commit of streams

* refactor stream component

* refactor so stream formats are not considered a platform

* initial test and minor refactor

* fix linting

* update requirements

* need av in tests as well

* fix import in class def vs method

* fix travis and docker builds

* address code review comments

* fix logger, add stream start/stop logs, listen to HASS stop

* address additional code review comments

* beef up tests

* fix tests

* fix lint

* add stream_source to onvif camera

* address pr comments

* add keepalive to camera play_stream service

* remove keepalive and move import

* implement registry and have output provider remove itself from stream after idle, set libav log level to error

robbiet480 committed:
Update lametric icon to be HA logo (#21957)

amelchio committed:
Avoid playing queue pollution when restoring Sonos snapshots (#21963)

Assume a snapshot state with three speakers in two groups, AB and C. They will
be playing the A and C queues, respectively. The B queue exists but is hidden
in this topology.

Unjoin B and form a new group BC, playing the B queue (now with the C queue
hidden).

To restore the snapshot we would join B back to A. The BC group would now only
contain the C speaker, still playing the B queue. The C queue has been lost :-(

The problem is that unjoining a coordinator will elect a new coordinator that
inherits the group queue and thus has its hidden queue overwritten.

This commit avoids the situation by having restore unjoin all slaves. Above, C
would be unjoined before joining B to A. This restores the C queue and since B
is then alone, it can be joined to A without having to transfer its playing
queue to remaining speakers.

kbickar committed:
Update to sense component to fully be async (#21698)

* Update to sense component to fully be async

* Shortened lines

* Whitespace

ttroy50 committed:
fix ephember doing http call from property (#21855)

Commit:
Add Weather Sensors to Homematic IP (#21887)

* Add HmIP Weather Sensor Devices

* Fix test and icons

* fix test

* Fix comments

balloob committed:
Add cloud status (#21960)

* Add cloud status

* Expose certificate details

* store & reset last state

* Fix tests

* update tests

* update req

* fix lint

Commit:
Fix MagicHome LEDs with flux_led component (#20733)

* bug fixing for MagicHome LEDs with flux_led component.

* corrections of the fixes for flux_led

* now asyncio sleep and turn on with brigthness possible

* indention fix with flux_led

* async now works

* houndci fixes

* little fixes for flux_led

* self._color fix for flux_led

* Add docstring

balloob committed:
Updated frontend to 20190312.0

balloob committed:
Update translations

balloob committed:
Stream: Only add base url when needed (#21979)

balloob committed:
Allow changing password without being admin (#21978)

balloob committed:
Fix some cloud things (#21977)

balloob committed:
Bumped version to 0.89.2

balloob committed:
Merge pull request #21982 from home-assistant/rc

0.89.2

Jc2k committed:
homekit_controller: Bump homekit to 0.13.0 (#21965)

* Bump homekit to 0.13.0

* Update gen_requirements_all.py

* Escape values used in TESTS_REQUIREMENTS

balloob committed:
skip flaky test (#21981)

emontnemery committed:
Write state directly in all MQTT platforms (#21971)

robbiet480 committed:
Use .get to ensure we dont get KeyError (#21993)

Jc2k committed:
Fix error introduced by #21933 (#21988)

dshokouhi committed:
Add stream source for amcrest component (#21983)

Jc2k committed:
Set homekit controller entity as unavailable if new connections fail (#21901)

* Set entity as unavailable if new connections fail

* Fix docstring

dgomes committed:
Centrally define Energy Units (kWh and Wh) (#21719)

* centralize energy units kWh and Wh

* lint

robbiet480 committed:
Mobile App: Registration schema improvements (#21850)

* Update registration schema to add os_name (required) and make app_name required

* Ensure that a provided app_component is valid and available

* Ensure that component DEPENDENCIES declares mobile_app

* Update homeassistant/helpers/config_validation.py

* Standardize error responses

* Dont generalize REGISTER_BAD_COMPONENT

* Fix tests after merge

balloob committed:
Add a service require_admin wrapper (#21953)

* Add a service require_admin wrapper

* Allow it to be used as a decorator

* Lint

* Add comment

* Add docstring

* Update syntax

danielperna84 committed:
Update pyhomematic to 0.1.58 (#21989)

amelchio committed:
Use asyncio lock (#21985)

awarecan committed:
Add breaking change section to PR template (#21994)

JeffLIrion committed:
Rename 'firetv' to 'androidtv' and add Android TV functionality (#21944)

* Working on adding androidtv functionality to firetv component

* 'should_poll' must return True

* Change 'properties' to 'device_properties'

* Also mention 'Android TV' in services.yaml

* Use GitHub for 'androidtv' requirement

* Add 'androidtv==0.0.10' to requirements, remove 'firetv==1.0.9'

* Add 'GET_PROPERTIES' adb command option; use pypi for REQUIREMENTS

* Rename integration from 'firetv' to 'androidtv'

* Change default name to 'Android TV'

* Rename integration from 'firetv' to 'androidtv'

* Change firetv to androidtv in .coveragerc

* Change firetv to androidtv in requirements_all.txt

* Remove 'DEFAULT_APPS'

endor-force committed:
Tellstick sensor configuration cleanup (#21402)

* Cleaned up named sensor handling for future

Breaking change for tellstick sensor configuration.

* Fixed linting of long lines and closing bracket

Linting warning for long line and not using best practice for visual indentation of closing bracket.
Who's a good boy..

* Whitespace on line was not cleaned.

* Removed spaces to clean up

* More.. whitespace. Sloppy.

* Constants from const, altered loops, added dictionary for sensor names.

* Fixed whitespace

* Inverted condition and created guard clause

* Changed condition from not ... in to not in.

* Fixed bad indentation on L91 and 92

bachya committed:
Add availability and next run datetime to RainMachine switches (#21786)

bachya committed:
Add program/zone enable/disable services to RainMachine (#21785)

pvizeli committed:
Bump NabuCasa library to 0.5 (#22010)

balloob committed:
Updated frontend to 20190313.0

balloob committed:
Update translations

amelchio committed:
Avoid playing queue pollution with Sonos unjoin (#22004)

amelchio committed:
Wait for Sonos regrouping in service calls (#22006)

beavis9k committed:
fix error in LutronButton init if Button doesn't have a type (#21921)

Jc2k committed:
HomeKit controller light - remove code that can never execute (#21951)

Oro committed:
Nanoleaf availability check (#21945)

* Added availability check for nanoleaf lights

* pylint errors fixed

* pynanoleaf bump

emontnemery committed:
Pass Message object to MQTT message callbacks (#21959)

* Pass Message object to MQTT message callbacks

* Improve method of detecting deprecated msg callback

* Fix mysensors

* Fixup

* Review comments

* Fix merge error

andrewsayre committed:
Remove default temp unit (#22012)

isabellaalstrom committed:
Allow all success status codes in REST notify response (#22011)

For example Discord webhooks returns a 204 success code as response, which gets logged as an error in the log, even though it is successful.
Update the allowed statuses to accept all 2xx responses as successful.

GGeezes committed:
Check updated_date for list and pick first (#22008)

balloob committed:
Merge branch 'master' into dev

robbiet480 committed:
Mobile App: Require encryption for registrations that support it (#21852)

## Description:

**Related issue (if applicable):** fixes #21758

## Checklist:
  - [x] The code change is tested and works locally.
  - [x] Local tests pass with `tox`. **Your PR cannot be merged unless tests pass**
  - [x] There is no commented out code in this PR.

robbiet480 committed:
Mobile App: Support rendering multiple templates at once (#21851)

* Support rendering multiple templates at once

* Only catch TemplateError and dont log the error

robbiet480 committed:
Mobile App: Update Location schema updates & device ID generation (#21849)

* Update location schema

* Generate a random device ID at registration time for later use with device_tracker.see

* Remove host name from device_tracker.see payload

* Drop consider_home from the payload

* Remove stale consider_home in schema

* Remove source_type

philhawthorne committed:
Remove UTF8 decoding for Waze (#22020)

Removes the UFT8 decoding for the Waze sensor, which broke in 0.89

Fixes #21739

robbiet480 committed:
Mobile App: Enable loading via discovery (surprise inside!) (#22027)

![](http://funpeep.com/wp-content/uploads/2014/04/Cute-White-Cat-Wallpaper.jpg)

robbiet480 committed:
Mobile App: Discovery to default configuration.yaml, zeroconf to default_config (#22028)

* Move discovery into default configuration.yaml

* Add zeroconf to default_config

robbiet480 committed:
Add myself to CODEOWNERS

JeffLIrion committed:
Bump androidtv to 0.0.11 (#22025)

kbickar committed:
Add sense guard clause (#22014)

awarecan committed:
Upgrade prefetched tox env to py37 (#22029)

mvn23 committed:
Bump pyotgw to 0.4b2 (#21973)

Danielhiversen committed:
change logging to debug for nmap (#22036)

dmulcahey committed:
Refactor ZHA gateway into modules and add admin protections to API (#22023)

* refactor

* cleanup

* fix tests

* admin all the things

awarecan committed:
Fix lifx light async error (#22031)

ktnrg45 committed:
Ps4 remove throttling (#21961)

* Remove throttling

* Correct docstring

fabaff committed:
Upgrade youtube_dl to 2019.03.09 (#22041)

cgtobi committed:
Bring back the boiler status (#22021)

fredrike committed:
Async support for Daikin (#21638)

* asyncio support for Daikin

* version bump pydaikin

* pass session to pydaikin.Appliance

* all entities should have update

pnbruckner committed:
Bump amcrest to 1.2.6 & use new exceptions (#22040)

* Bump amcrest to 1.2.6 & use new exceptions

* Fix lint failure

* Use AmcrestError instead of individual ones

emontnemery committed:
Update additional platforms to use new MQTT message callback (#22030)

* Move additional platforms to new MQTT callback

* Fix automation.mqtt

cgarwood committed:
Z-Wave usb_path in configuration.yaml overrides config entry usb_path (#22038)

* usb_path in configuration.yaml overrides config entry

* Minor text update

* Update __init__.py

robbiet480 committed:
Mobile App: Register devices into the registry (#21856)

* Register devices into the registry

* Switch to device ID instead of webhook ID

* Rearchitect mobile_app to support config entries

* Kill DATA_REGISTRATIONS by migrating registrations into config entries

* Fix tests

* Improve how we get the config_entry_id

* Remove single_instance_allowed

* Simplify setup_registration

* Move webhook registering functions into __init__.py since they are only ever used once

* Kill get_registration websocket command

* Support description_placeholders in async_abort

* Add link to mobile_app implementing apps in abort dialog

* Store config entry and device registry entry in hass.data instead of looking it up

* Add testing to ensure that the config entry is created at registration

* Fix busted async_abort test

* Remove unnecessary check for entry is None

pvizeli committed:
Fix Google Assistant User with Cloud (#22042)

* Fix Google Assistant User with Cloud

* Fix User Agent ID

* respell

* Fix object

* Fix tests

* fix lint

* Fix lint

robbiet480 committed:
Mobile App: Sensors (#21854)

## Description:

**Related issue (if applicable):** fixes #21782

## Checklist:
  - [x] The code change is tested and works locally.
  - [x] Local tests pass with `tox`. **Your PR cannot be merged unless tests pass**
  - [x] There is no commented out code in this PR.

Adminiuga committed:
Minor ZHA consts fixes (#22049)

* Use consts for zha.core.consts

* Fix ZHA battery reporting config.

robbiet480 committed:
Fix cloud services.yaml

dmulcahey committed:
Cleanup remaining constants stuff in ZHA (#22050)

* clean up constants
* fix quirks until it can be upgradded

robbiet480 committed:
Mobile App: Expose Cloud Remote UI FQDN in registration response (#22055)

* Add a callback to get the cloud remote UI FQDN

* Expose Cloud Remote UI FQDN in the registration response

* Return a URL instead of FQDN

escoand committed:
Add HTTP auth and SSL verification to REST notify (#22016)

* add HTTP auth and SSL verification

* use internal import

* fix long line

* avoid extra import

balloob committed:
Return config entry ID after creation (#22060)

balloob committed:
Fix test

balloob committed:
Fix more test

fabaff committed:
Upgrade ruamel.yaml to 0.15.89 (#22064)

awarecan committed:
Correct context (#22061)

isabellaalstrom committed:
Update rest.py (#22077)

Added specific error logs for 5xx and 4xx responses, debug log for 2xx and other statuses.

pvizeli committed:
Binary Sensor for Remote UI & Fix timezone (#22076)

* Binary Sensor for Remote UI

* Fix lint

* Revert make hass public

* Add tests

balloob committed:
Updated frontend to 20190315.0

balloob committed:
Fix func

JeffLIrion committed:
Bump androidtv to 0.0.12 (#22072)

nbarrientos committed:
Telegram_bot: Allow fetching data from unverified SSL endpoints (#22067) (#22069)

* Telegram_bot: Allow fetching data from unverified SSL endpoints (#22067)

This patch adds an extra option to the payload that describes the
resource to send to disable the SSL server verification when polling
data from URLs.

* Use the boolean interpretation of the variable directly

Co-Authored-By: nbarrientos <nacho@criptonita.com>

balloob committed:
Prevent cloud remote UI when using 127.0.0.1 as trusted network (#22093)

* Prevent cloud remote UI when using trusted networks

* Limit to 127.0.0.1 trusted network

* Update error msg

* Disable ipv6 loopback

alengwenus committed:
Add LCN sensor platform (#21440)

balloob committed:
Updated frontend to 20190315.1

hunterjm committed:
Beta Fix: FFMPEG and Stream component (#22091)

* remove stream_source from ffmpeg and onvif and add to generic ip cam

* fix tests

balloob committed:
Version bump to 0.91.0dev0

scop committed:
Upgrade pytest to 4.3.1 (#22088)

pnbruckner committed:
Bump amcrest to 1.2.7 for correct RTSP port (#22099)

amcrest 1.2.7 now includes camera's configured RTSP port in generated URL. Necessary to make new stream component work correctly if camera does not use default RTSP port.

bachya committed:
Add available property to Ambient PWS (#22092)

* Add available property to Ambient PWS

* Linting

* remove blank line

alengwenus committed:
Restricted temperature units for Celsius and Fahrenheit to the standardized units (#22108)

mvn23 committed:
Fix opentherm_gw blocks HA startup when gateway unreachable. (#22106)

mvn23 committed:
Fix TypeError in current_temperature if no temperature is known. (#22112)

Don't set opentherm_gw climate temperatures to 0 on init.

mvn23 committed:
Improve opentherm gw startup (#22121)

* Improve fix in c90f0d5 (#22106). Schedule connect coroutine directly on the loop rather than waiting for EVENT_HOMEASSISTANT_START.

* Remove unused import.

balloob committed:
Consolidate all platforms that have no tests (#22096)

* Consolidate

* Fix tests

* Update imports

* Fix import

* Use importlib because integration and package share name

* Fix more tests

* Update .coveragerc and CODEOWNERS

hunterjm committed:
delete previously removed service option from services yaml (#22123)

balloob committed:
Updated frontend to 20190316.0

OttoWinter committed:
Add ESPHome Camera (#22107)

* Add ESPHome Camera

* Bump aioesphomeapi to 1.7.0

Danielhiversen committed:
Tibber, Add price level (#22085)

* Add price level to Tibber sensor

* bump pyTibber version

awarecan committed:
Remove hass.config from aws_lambda notify payload (#22125)

iamtpage committed:
Fix AttributeError traceback with darksky (#22101)

* Update darksky.py

* fixes

cgtobi committed:
Add service description for services. (#21897)

* Add service description for services.

* Make service description more clear

bachya committed:
Handle occassional 502 Bad Gateway from SimpliSafe (#22131)

bachya committed:
Loosen field retrieval from Pollen.com (#22132)

bachya committed:
Handle possible exception during Ambient PWS reconnect (#22134)

* Handle possible exception during Ambient PWS reconnect

* Hound

balloob committed:
Allow non-admins to listen to certain events (#22137)

WebSpider committed:
Bump tado version (#22145)

* Bump python-tado, new API endpoint

* Change references of old API endpoint to new

* Update REQUIREMENTS

hunterjm committed:
Fix resetting access token on streams with keepalive (#22148)

endor-force committed:
Additional sensors and fixed icons, attributes and device classes. (#22139)

jameshilliard committed:
Fix hlk-sw16 inverted switch state (#22059)

benleb committed:
replace einder lib with horimote (#22135)

zewelor committed:
Fixx yeelight flow action param for declared effects (#22159)

balloob committed:
Updated frontend to 20190318.0

frenck committed:
Upgrade toonapilib to 3.2.2 + lower interval (#22160)

pvizeli committed:
Remove config check over supervisor (#22156)

* Remove config check over supervisor

* Fix lint

* Fix tests

dmulcahey committed:
Add dynamic subscription for ZHA add device page (#22164)

* add ws subscription for zha gateway messages
* add debug mode
* only relay certain logs
* add missing require admin
* add devices command
* add area_id
* fix manufacturer code

Swamp-Ig committed:
Consolidate all platforms that have tests (#22109)

* Moved climate components with tests into platform dirs.

* Updated tests from climate component.

* Moved binary_sensor components with tests into platform dirs.

* Updated tests from binary_sensor component.

* Moved calendar components with tests into platform dirs.

* Updated tests from calendar component.

* Moved camera components with tests into platform dirs.

* Updated tests from camera component.

* Moved cover components with tests into platform dirs.

* Updated tests from cover component.

* Moved device_tracker components with tests into platform dirs.

* Updated tests from device_tracker component.

* Moved fan components with tests into platform dirs.

* Updated tests from fan component.

* Moved geo_location components with tests into platform dirs.

* Updated tests from geo_location component.

* Moved image_processing components with tests into platform dirs.

* Updated tests from image_processing component.

* Moved light components with tests into platform dirs.

* Updated tests from light component.

* Moved lock components with tests into platform dirs.

* Moved media_player components with tests into platform dirs.

* Updated tests from media_player component.

* Moved scene components with tests into platform dirs.

* Moved sensor components with tests into platform dirs.

* Updated tests from sensor component.

* Moved switch components with tests into platform dirs.

* Updated tests from sensor component.

* Moved vacuum components with tests into platform dirs.

* Updated tests from vacuum component.

* Moved weather components with tests into platform dirs.

* Fixed __init__.py files

* Fixes for stuff moved as part of this branch.

* Fix stuff needed to merge with balloob's branch.

* Formatting issues.

* Missing __init__.py files.

* Fix-ups

* Fixup

* Regenerated requirements.

* Linting errors fixed.

* Fixed more broken tests.

* Missing init files.

* Fix broken tests.

* More broken tests

* There seems to be a thread race condition.
I suspect the logger stuff is running in another thread, which means waiting until the aio loop is done is missing the log messages.
Used sleep instead because that allows the logger thread to run. I think the api_streams sensor might not be thread safe.

* Disabled tests, will remove sensor in #22147

* Updated coverage and codeowners.

Commit:
Change ATTR_BATTERY_STATE_UNPLUGGED for ios component (#22152)

* AH - Updated Text Description of ATTR_BATTERY_STATE_UNPLUGGED

* AH - updated device_tracker ios test to use new battery state description

uchagani committed:
bump total_connect_client to 0.24 (#22166)

Jc2k committed:
Add some tests for Koogeek LS1 (#22141)

Swamp-Ig committed:
Move platforms patch (#22168)

* move mold sensor

* Get tests working

* Move moon and season sensor strings.

* Renamed sensor strings.

pvizeli committed:
Update Hass-NabuCasa 0.8 (#22177)

Danielhiversen committed:
restore entity for switchbot (#22087)

eliseomartelli committed:
removed iliad (#22175)

panosmz committed:
move oasa_telematics component to own folder

panosmz committed:
Merge remote-tracking branch 'origin/oasa-telematics' into oasa-telematics

robbiet480 commented:
This is a PR gone wrong, locking it as such.

robbiet480 commented:
Oh yeah @panosmz feel free to submit a new _clean_ PR obviously.
